### PR TITLE
Custom cmd

### PR DESCRIPTION
**Pull request by rsevilla87**
### Description

Add logic to trigger custom commands rather than running only e2e-benchmarking based workloads. This will provide us much flexibility and freedom to include new benchmarks. 
Also updating base image to include latest kube-burner binary and adding some temporally example dags for testing purposes.

### CODE REVIEW

File: dags/openshift_nightlies/tasks/utils/final_dag_status.py [hunk 1]
```diff
@@ -23,7 +23,6 @@ def final_status(**kwargs):
 def get_task(dag):
     return PythonOperator(
         task_id='final_status',
-        provide_context=True,
```
Comment (rsevilla87) [inline dags/openshift_nightlies/tasks/utils/final_dag_status.py:26]: This argument is deprecated, removing it.

File: dags/openshift_nightlies/scripts/run_benchmark.sh [hunk 2]
```diff
@@ -78,12 +81,16 @@ echo "############################################"
...
     run_baremetal_benchmark
-    echo $UUID
 else
     setup
-    cd /home/airflow/workspace/e2e-benchmarking/workloads/$workload
+    if [[ -n ${workload} ]]; then
+        echo "Cloning ${E2E_BENCHMARKING_REPO} from branch ${E2E_BENCHMARKING_BRANCH}"
+        git clone -q -b ${E2E_BENCHMARKING_BRANCH} ${E2E_BENCHMARKING_REPO} --depth=1 --single-branch
+        cd /home/airflow/workspace/e2e-benchmarking/workloads/$workload
+    fi
```
Comment (rsevilla87) [inline dags/openshift_nightlies/scripts/run_benchmark.sh:90]: Only clone e2e-benchmarking repository when a specific workload is passed.

File: dags/openshift_nightlies/config/benchmarks/control-plane-custom.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            "env": {
+                "SCALE": "27",
+                "METADATA_COLLECTION": "true",
+                "WORKLOAD_NODE_ROLE": "workload"
+            }
+        },
+        {
+            "name": "node-density",
+            "workload": "kube-burner",
+            "custom_cmd": "kube-burner ocp node-density --uuid=${UUID} --pods-per-node=245 --timeout=2h --pod-ready-threshold=5s --es-server=${ES_SERVER} --es-index=${ES_INDEX}"
```
Comment (rsevilla87) [inline dags/openshift_nightlies/config/benchmarks/control-plane-custom.json:16]: The variables UUID, ES_SERVER and ES_INDEX are already automatically injected by airflow

File: dags/openshift_nightlies/scripts/run_benchmark.sh [hunk 4]
```diff
@@ -97,6 +104,5 @@ else
         export WORKLOAD=$AIRFLOW_CTX_TASK_ID-must-gather
         ./ocp_diagnosis.sh > /dev/null
     fi
-    echo $UUID
```
Comment (mukrishn) [inline dags/openshift_nightlies/scripts/run_benchmark.sh:100]: was this intentional ? I think we need this for [indexing](https://github.com/cloud-bulldozer/airflow-kubernetes/blob/master/dags/openshift_nightlies/tasks/index/status.py#L51) job, unless if its printed by e2e or kube-burner scripts.

Comment (rsevilla87) [reply dags/openshift_nightlies/scripts/run_benchmark.sh:100]: woops, good catch, I wasn't aware of this trick, reverting this change

File: dags/openshift_nightlies/tasks/utils/final_dag_status.py (hunk 1 above)
Comment (Harshith-umesh) [reply dags/openshift_nightlies/tasks/utils/final_dag_status.py:26]: @rsevilla87 this is required as we need to pass airflow context variables to kwargs of final_status.

Comment (rsevilla87) [reply dags/openshift_nightlies/tasks/utils/final_dag_status.py:26]: > @rsevilla87 this is required as we need to pass airflow context variables to kwargs of final_status.

https://airflow.apache.org/docs/apache-airflow/2.3.1/release_notes.html?highlight=provide_context#airflow-operators-python-pythonoperator

Comment (Harshith-umesh) [reply dags/openshift_nightlies/tasks/utils/final_dag_status.py:26]: ah okay!

**Review by Harshith-umesh — APPROVED** [review]
LGTM!

**Pull request merged**